src/storage/v2/edge_accessor.cpp

```diff
@@ -70,51 +68,112 @@ bool EdgeAccessor::IsVisible(const View view) const {
         case Delta::Action::DELETE_DESERIALIZED_OBJECT:
         case Delta::Action::DELETE_OBJECT:
           break;
-        case Delta::Action::ADD_OUT_EDGE: {  // relevant for the from_vertex_ -> we just deleted the edge
+        case Delta::Action::ADD_OUT_EDGE: {
+          if (delta.vertex_edge.edge == edge_ && delta.vertex_edge.edge_type == edge_type_) {
+            attached = true;
+          }
+          break;
+        }
+        case Delta::Action::REMOVE_OUT_EDGE: {
           if (delta.vertex_edge.edge == edge_) {
-            deleted = false;
+            attached = false;
+          }
+          break;
+        }
+      }
+    });
+    return attached;
+  };
+  auto check_to_vertex_integrity = [&view, this]() -> bool {
+    bool attached = true;
+    Delta *delta = nullptr;
+    {
+      auto guard = std::shared_lock{to_vertex_->lock};
+      // Initialize deleted by checking if out edges contain edge_
+      attached = std::find_if(to_vertex_->in_edges.begin(), to_vertex_->in_edges.end(), [&](const auto &in_edge) {
+                   return std::get<2>(in_edge) == edge_ && std::get<0>(in_edge) == edge_type_;
+                 }) != to_vertex_->in_edges.end();
+      delta = to_vertex_->delta;
+    }
+    ApplyDeltasForRead(transaction_, delta, view, [&](const Delta &delta) {
+      switch (delta.action) {
+        case Delta::Action::ADD_LABEL:
+        case Delta::Action::REMOVE_LABEL:
+        case Delta::Action::SET_PROPERTY:
+        case Delta::Action::REMOVE_OUT_EDGE:
+        case Delta::Action::ADD_OUT_EDGE:
+        case Delta::Action::RECREATE_OBJECT:
+        case Delta::Action::DELETE_DESERIALIZED_OBJECT:
+        case Delta::Action::DELETE_OBJECT:
+          break;
+        case Delta::Action::ADD_IN_EDGE: {
+          if (delta.vertex_edge.edge == edge_ && delta.vertex_edge.edge_type == edge_type_) {
+            attached = true;
           }
           break;
         }
-        case Delta::Action::REMOVE_OUT_EDGE: {  // also relevant for the from_vertex_ -> we just added the edge
+        case Delta::Action::REMOVE_IN_EDGE: {
           if (delta.vertex_edge.edge == edge_) {
-            exists = false;
+            attached = false;
           }
           break;
         }
       }
     });
+    return attached;
+  };
+  auto check_presence_of_edge = [&view, this]() -> bool {
+    bool deleted = true;
+    bool exists = true;
```

Comment:
Do we need two bools to define this?
Is one not enough?

Reply:
I just left it from previous code, they were using 2 bools and use 2 bools in the return check

Reply:
pushed one bool instead of 2